### DISCUSSION TITLE
Add UI for when reviews are in progress

### DISCUSSION THREAD

**Discussion by mehulkar** [community/community#7552] [Pull Requests]
It would be useful to know when someone has started a review (i.e. has locally started adding comments, but the comments are not visible to others yet). If I knew this (via some UI), the PR author could be more careful about adding new commits, rebasing or otherwise changing the PR. 

From https://github.com/isaacs/github/issues/1137

**Comment by jglick** [comment]
#9711 would be a sort of workaround.

Sent to private support a year ago:

Sometimes it happens that someone files a PR, requests review from a few people, and gets a perfunctory approval from one person without deep knowledge of the affected code. Meanwhile, someone else has seen the review request and is beginning a detailed review that could take an hour to complete. The author knows nothing about that, sees the approval, and merges. The detailed reviewer winds up submitting their review on a PR which has already been merged, forcing a much more awkward workflow of asking for a revert, or a follow-up PR to correct minor issues.

As a workaround, sometimes I try to react to a PR with :eyes: but the author (or whoever is responsible for merging) may or may not notice that. Adding a global comment to the end of the PR to the effect of

> Hey hold on a bit, I am reviewing this!

is more prominent, but noisy. Or sometimes I send a note out of band, say in Slack. It is easy to forget to do these things, though, and then if you get distracted by some other task and forget about the PR, the status could be unclear for days.

I would like there to be some notification (visible at least to people with write access, and thus authorized to merge the PR, and maybe also to the author) that one or more reviews are still being drafted—like how the reviewer sees their own draft line comments highlighted in yellow if they visit the PR’s main tab. Preferably such a notification would be displayed near the merge button, since the top of the PR’s main tab may be scrolled offscreen.

Additionally / alternatively, it would be nice to display some indication that people are looking at the PR (especially the Files tab), even if they have not yet written any draft review comments. Google Docs, for example, shows avatars of other users with a document open. For security reasons this kind of display should I suppose be shown only to people with write access, and probably only display viewers who also have write access and thus could submit binding reviews.

**Comment by wtross-eg** [comment]
Mentioning a UI is a bit misleading. What is needed is to have the information available that someone has _started_ a review. How to make it available is up to whoever implements it. It could be the reviewer appearing in the reviewers' box (to the right of the PR) with some mark next to them, or other visual feedback.
This is particularly important in enterprise situations where the PR owner waits for the minimum number of approvals, and then merges the PR if no comments are pending.
**In my current team we decided to use labels like `John reviewing`, but there should be a better way.**

**Comment by salomoneb** [comment]
I agree with the comments above and would love to see this feature. I also don't care about having a fancy UI so much as knowing that someone is looking at the PR. There are all sorts of scenarios where reviews are delayed: the reviewer is away, the PR is complex, other things are more important, etc, etc. Having some indicator that the PR is being reviewed communicates that the team/reviewer has not forgotten about it and is prioritizing it without the author needing to do extra prodding.

**Comment by calvin-cdev** [comment]
Please do this. I find myself mid-review when the icon at the top turns purple with "Merged" far too often.

**Reply by JonathanVanDyke** [reply]
Seconded / bump
